src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/JCloudsProvider.java

```diff
@@ -143,14 +147,16 @@ public void deleteInstance(Infrastructure infrastructure, String instanceId) {
             InstanceScript instanceScript) {
 
         Map<? extends NodeMetadata, ExecResponse> execResponses;
-
+        RunScriptOptions runScriptOptions = null;
         try {
+            String scriptToExecuteString = buildScriptToExecuteString(instanceScript);
+            runScriptOptions = buildScriptOptionsWithInstanceTag(instanceScript, instanceTag, infrastructure);
             execResponses = getComputeServiceFromInfastructure(infrastructure).runScriptOnNodesMatching(runningInGroup(instanceTag),
-                                                                                                        buildScriptToExecuteString(instanceScript),
-                                                                                                        buildScriptOptionsWithInstanceTag(instanceScript,
-                                                                                                                                          instanceTag,
-                                                                                                                                          infrastructure));
+                                                                                                        scriptToExecuteString,
+                                                                                                        runScriptOptions);
         } catch (Exception e) {
+            log.error("Script cannot be run on instance with tag: " + instanceTag + ". RunScriptOptions=" +
+                      runScriptOptions, e);
             throw new RuntimeException("Script cannot be run on instances with tag: " + instanceTag, e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)